doc/languages-frameworks/vim.section.md

```diff
@@ -170,8 +170,8 @@ of precompiled grammars, you can use `nvim-treesitter.withPlugins` function:
       start = [
         (nvim-treesitter.withPlugins (
           plugins: with plugins; [
-            tree-sitter-nix
-            tree-sitter-python
+            nix
+            python
```

Comment:
I just updated and this syntax did not break. Is this expected? Am I mixing tree-sitter files with https://github.com/NixOS/nixpkgs/blob/nixos-unstable/pkgs/development/tools/parsing/tree-sitter/grammar.nix#L21 ?

Reply:
Yes, like I thought. If you are using ``with vimPlugins; []`` and do not notice this change (which I expect the majority will not) then the vimPlugins tree-sitter packages I linked in the above comment are used and currently you get all kinds of errors. Can we fix this better so that this mixing is not possible?

Reply:
it was intended for existing configurations to not change, since the naming schemes are different, maybe we can add aliases to append `tree-sitter` and change `_` to `-` but I don't think it's possible to change the behavior of `nvim-treesitter.withPlugins (_: tree-sitter.allGrammars)`, which is what i was using and what i assume a lot of other people are using as well

Reply:
Can we somehow detect if the wrong packages are being used inside nix and then throw?

Reply:
there are definitely ways but im not sure if this is going to be worth it, since there might still be reasons to use the grammars from tree-sitter for example if its not included in nvim-treesitter